pkg/mtu/detect_linux.go

```diff
@@ -75,3 +75,33 @@ func autoDetect() (int, error) {
 
 	return EthernetMTU, nil
 }
+
+// getMTUFromIf finds the interface that holds the ip and returns its mtu
+func getMTUFromIf(ip net.IP) (int, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return 0, fmt.Errorf("Unable to list interfaces: %s", err)
+	}
+
+	for _, iface := range ifaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return 0, fmt.Errorf("Unable to list the addresses: %s", err)
+		}
+
+		for _, addr := range addrs {
+			myIP, _, err := net.ParseCIDR(addr.String())
+
+			if err != nil {
+				return 0, fmt.Errorf("Unable to parse the CIDR: %s", err)
+			}
+
+			if myIP.Equal(ip) == true {
+				myMTU := iface.MTU
+				log.Infof("External network interface uses MTU %d. Cilium will use that MTU too", myMTU)
```

Comment:
Please format these logs with all of the info about how this MTU was detected, using the `WithFields()` syntax as that's friendlier to log parsers. Example (note some `logfields` constants might not exist from here, you may need to add them):

```suggestion
				log.WithFields(
				  logfields.Device: iface.Name(),
				  logfields.IPAddr: ip,
				  logfields.MTU: myMTU,
				).Info("Inheriting MTU from external network interface")
```

Reply:
Ah, thanks! This is a good idea